tests/integration/test_external_url.py

```diff
@@ -0,0 +1,99 @@
+"""
+
+import asyncio
+import logging
+import subprocess
+import urllib.request
+
+import pytest
+from helpers import oci_image
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+prometheus_app_name = "prometheus"
+prometheus_resources = {"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")}
+
+# Two prometheus units are sufficient to test potential interactions between multi-unit
+# deployments and external_url
+num_units = 2
+
+# The period of time required to be idle before `wait_for_idle` returns is set to 90 sec because
+# the default scrape_interval in prometheus is 1m.
+idle_period = 90
+
+# Set up microk8s metallb addon, needed by traefik
+cmd = [
+    "sh",
+    "-c",
+    "ip -4 -j route | jq -r '.[] | select(.dst | contains(\"default\")) | .prefsrc'",
+]
+result = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+ip = result.stdout.decode("utf-8").strip()
+
+cmd = ["sg", "microk8s", "-c", f"microk8s enable metallb:{ip}-{ip}"]
+subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+
+async def test_setup_env(ops_test: OpsTest):
+    await ops_test.model.set_config(
+        {"logging-config": "<root>=WARNING; unit=DEBUG", "update-status-hook-interval": "60m"}
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_self_scrape_job_is_up_with_traefik(ops_test: OpsTest, prometheus_charm):
+    await asyncio.gather(
+        ops_test.model.deploy(
+            prometheus_charm,
+            resources=prometheus_resources,
+            application_name=prometheus_app_name,
+            num_units=num_units,
+            trust=True,
+        ),
+        ops_test.model.deploy(
+            "ch:traefik-k8s",
+            application_name="traefik",
+            channel="edge",
+        ),
+    )
+
+    app_names = [prometheus_app_name, "traefik"]
+
+    await asyncio.gather(
+        # ops_test.model.applications["traefik"].set_config(
+        #     {"alert-rules-path": "src/with_extra_alert_rule"}
+        # )
+        ops_test.model.add_relation(f"{prometheus_app_name}:ingress", "traefik"),
+        ops_test.model.wait_for_idle(
+            apps=app_names,
+            status="active",
+            wait_for_units=num_units,
+            timeout=600,
+            idle_period=idle_period,
+        ),
+    )
+
+    def prom_url(unit: int):
+        return f"http://{ip}/{ops_test.model}-{prometheus_app_name}-{unit}"
+
+    prom_urls = [prom_url(i) + "/api/v1/targets" for i in range(num_units)]
+    logger.info("Attmpting to fetch targets from these urls: %s", prom_urls)
+    targets_per_unit = [
+        urllib.request.urlopen(url, None).read().decode("utf8") for url in prom_urls
+    ]
+    assert all('"health":"down"' not in response for response in targets_per_unit)
+
+
+@pytest.mark.abort_on_fail
+async def test_self_scrape_job_is_up_with_config_option_overriding_traefik(ops_test: OpsTest):
+    pass
```

Comment:
I think you forgot to remove this placeholder.

Reply:
Done.